fission/src/test/scene/SceneRenderer.test.ts

```diff
@@ -0,0 +1,509 @@
+
+            expect(worldPos.x).toBe(0)
+            expect(worldPos.y).toBe(0)
+        })
+
+        test("should convert world to pixel space", () => {
+            const pixelPos = sceneRenderer.worldToPixelSpace(new THREE.Vector3(0, 0, 0))
+            expect(pixelPos[0]).toBeCloseTo(1861, 0)
+            expect(pixelPos[1]).toBeCloseTo(1261, 0)
+        })
+    })
+
+    describe("Canvas Management", () => {
+        test("should update camera aspect ratio based on window size", () => {
+            // Windows size is already set to 1920x1080
+            sceneRenderer.updateCanvasSize()
+
+            const aspectRatio = 1920 / 1080
+            expect(sceneRenderer.mainCamera.aspect).toBeCloseTo(aspectRatio)
+            expect(sceneRenderer.mainCamera.fov).toBeCloseTo(110 / aspectRatio)
+        })
+
+        test("should handle wide aspect ratios correctly", () => {
+            Object.defineProperty(window, "innerWidth", { value: 3840 })
+            Object.defineProperty(window, "innerHeight", { value: 1080 })
+
+            sceneRenderer.updateCanvasSize()
+
+            const aspectRatio = 3840 / 1080
+            expect(sceneRenderer.mainCamera.aspect).toBeCloseTo(aspectRatio)
+
+            expect(sceneRenderer.mainCamera.fov).toBeCloseTo(110 / aspectRatio)
+        })
+
+        test("should handle tall aspect ratios correctly", () => {
+            Object.defineProperty(window, "innerWidth", { value: 800 })
+            Object.defineProperty(window, "innerHeight", { value: 1200 })
+
+            sceneRenderer.updateCanvasSize()
+
+            expect(sceneRenderer.mainCamera.aspect).toBeCloseTo(800 / 1200)
+            expect(sceneRenderer.mainCamera.fov).toBeCloseTo(110 / (16 / 9))
+        })
+    })
+
+    describe("Lighting", () => {
+        test("should switch between directional and CSM lighting modes", () => {
+            sceneRenderer.changeLighting(false)
+            const directionalLight = sceneRenderer.scene.children.find(child => child instanceof THREE.DirectionalLight)
+            expect(directionalLight).toBeInstanceOf(THREE.DirectionalLight)
+
+            sceneRenderer.changeLighting(true)
+            const noDirectionalLight = sceneRenderer.scene.children.find(
+                child => child instanceof THREE.DirectionalLight
+            )
+            expect(noDirectionalLight).toBeUndefined()
+
+            sceneRenderer.changeLighting(false)
+            const newDirectionalLight = sceneRenderer.scene.children.find(
+                child => child instanceof THREE.DirectionalLight
+            )
+            expect(newDirectionalLight).toBeInstanceOf(THREE.DirectionalLight)
+        })
+
+        test("should handle null material gracefully", () => {
+            sceneRenderer.changeLighting(true)
+
+            expect(() => sceneRenderer.setupMaterial(null as unknown as THREE.Material)).not.toThrow()
+        })
+    })
+
+    describe("Scene Management", () => {
```

Comment:
All of the Scene Management tests appear to exclusively test library code in THREE.Scene, not Synthesis code